core/templates/dev/head/galleries/Gallery.js

```diff
@@ -399,27 +401,39 @@ oppia.controller('Gallery', [
       }
     );
 
-    $scope.showFullGalleryGroup = function(galleryGroup) {
-      selectionDataService.clearCategories();
-      selectionDataService.addCategoriesToSelection(galleryGroup.categories);
-      // TODO(sll): is this line correct?
-      selectionDataService.clearLanguageCodes();
-      searchService.executeSearchQuery('', function() {
-        removeSplashCarousel();
-      });
-      // TODO(sll): Clear the search query from the search bar, too.
+    // Retrieves gallery data from the server.
+    $http.get(GALLERY_DATA_URL).success(function(data) {
```

Comment:
I'm not sure you need this method at all, since we are doing a GET of /default_gallery_categories above already.